src/work-allocation-2/containers/work-case-list-wrapper/work-case-list-wrapper.component.ts

```diff
@@ -0,0 +1,275 @@
+import { AppConstants } from '../../../app/app.constants';
+import { mergeMap } from 'rxjs/operators';
+import { SortField } from '../../models/common';
+
+@Component({
+  templateUrl: 'work-case-list-wrapper.component.html',
+  providers: [InfoMessageCommService]
+})
+export class WorkCaseListWrapperComponent implements OnInit {
+
+  public specificPage: string = '';
+  public caseworkers: Caseworker[];
+  public showSpinner$: Observable<boolean>;
+  public sortedBy: SortField;
+  public pagination: PaginationParameter;
+  public isPaginationEnabled$: Observable<boolean>;
+  private pCases: Case[];
+  /**
+   * Mock CaseServiceConfig.
+   */
+  private readonly defaultCaseServiceConfig: CaseServiceConfig = {
+    service: CaseService.IAC,
+    defaultSortDirection: SortOrder.ASC,
+    defaultSortFieldName: 'startDate',
+    fields: this.fields,
+  };
+
+  /**
+   * Take in the Router so we can navigate when actions are clicked.
+   */
+  constructor(
+    protected ref: ChangeDetectorRef,
+    protected caseService: WorkAllocationCaseService,
+    protected router: Router,
+    protected infoMessageCommService: InfoMessageCommService,
+    protected sessionStorageService: SessionStorageService,
+    protected alertService: AlertService,
+    protected caseworkerService: CaseworkerDataService,
+    protected loadingService: LoadingService,
+    protected featureToggleService: FeatureToggleService
+  ) {
+    this.isPaginationEnabled$ = this.featureToggleService.isEnabled(AppConstants.FEATURE_NAMES.waMvpPaginationFeature);
+  }
+
+  public get cases(): Case[] {
+    return this.pCases;
+  }
+
+  public set cases(value: Case[]) {
+    this.pCases = value;
+  }
+
+  private pCasesTotal: number;
+  public get casesTotal(): number {
+    return this.pCasesTotal;
+  }
+  public set casesTotal(value: number) {
+    this.pCasesTotal = value;
+  }
+
+  public get fields(): CaseFieldConfig[] {
+    return [];
+  }
+
+  public get caseServiceConfig(): CaseServiceConfig {
+    return this.defaultCaseServiceConfig;
+  }
+
+  public get emptyMessage(): string {
+    return ListConstants.EmptyMessage.Default;
+  }
+
+  /**
+   * To be overridden.
+   */
+  public get sortSessionKey(): string {
+    return 'sortSessionKey';
+  }
+
+  /**
+   * To be overridden.
+   */
+  public get view(): string {
+    return 'default';
+  }
+
+  public get returnUrl(): string {
+    return this.router ? this.router.url : '/mywork';
+  }
+
+  /**
+   * Flag to indicate whether or not we've arrived here following a bad
+   * request with a flag having been set on another route. The flag is
+   * passed through the router and so is held in window.history.state.
+   */
+  private get wasBadRequest(): boolean {
+    if (window && window.history && window.history.state) {
+      return !!window.history.state.badRequest;
+    }
+    return false;
+  }
+
+  public ngOnInit(): void {
+    this.setupCaseList();
+    this.loadCases();
+  }
+
+  public setupCaseList() {
```

Comment:
_Nitpick:_
Should have a return type of `void`.

Separately, it seems like a strange name for the method considering it deals with _caseworkers_ and not _cases_ at all.

Reply:
Returning a void now, and renamed to deal with caseworkers.